devops/providers/azure-devops/templates/azure-pipelines.yml

```diff
@@ -73,13 +74,15 @@ steps:
   displayName: 'Determine integration tests based off branch changes and load build path'
 
 - script: |
-      docker run -e ARM_SUBSCRIPTION_ID=$(AD-SP-SUBSCRIPTION-ID) \
-          -e ARM_CLIENT_ID=$(AD-SP-CLIENT-ID) \
-          -e ARM_CLIENT_SECRET=$(AD-SP-SECRET) \
-          -e ARM_TENANT_ID=$(AD-SP-TENANT-ID) \
-          -e DATACENTER_LOCATION=$DATACENTER_LOCATION \
-          -e TF_STATE_STORAGE_ACCT=$TF_STATE_STORAGE_ACCT \
-          -e TF_STATE_CONTAINER=$TF_STATE_CONTAINER \
-          -e ARM_ACCESS_KEY=$(ARM-ACCESS-KEY) \
-          --rm $BUILD_TEST_RUN_IMAGE:$BUILD_BUILDID
+    if test ! -z "$(docker images -q ${BUILD_TEST_RUN_IMAGE}:${BUILD_BUILDID})"; then
```

Comment:
When would the docker image not exist on the host? Do we want to fail in that case, or is there a use case I'm not aware of?

Also, TIL about `test`. Thanks!

Reply:
if there's no template directories to process then we wouldn't have built the image.